src/buf_writer.c

```diff
@@ -5,32 +5,128 @@
 #include "jemalloc/internal/buf_writer.h"
 #include "jemalloc/internal/malloc_io.h"
 
+static void *
+buf_writer_allocate_internal_buf(tsdn_t *tsdn, size_t buf_len) {
+	if (buf_len > SC_LARGE_MAXCLASS) {
```

Comment:
Since it’s internally used, an assertion will be fine.

Reply:
This was actually my trick: I had `SC_LARGE_MAXCLASS + 1` in my unit test for the purpose of explicitly triggering a OOM.

Reply:
I see. Let's either do a comment, or an #ifdef to only trigger in tests (and assert on non-tests)?

Reply:
Good point.